anoncreds/src/services/issuer.rs

```diff
@@ -169,58 +171,111 @@ where
-    };
-
-    // now update registry to reflect issuance-by-default
-    let (revoc_reg, revoc_init_delta) = {
-        let tails_reader = TailsFileReader::new_tails_reader(&tails_location);
-        let issued = BTreeSet::from_iter(1..=max_cred_num);
-        update_revocation_registry(
-            &revoc_reg_def,
-            &revoc_reg,
-            issued,
-            BTreeSet::new(),
-            &tails_reader,
-        )?
-    };
-
     let revoc_def_priv = RevocationRegistryDefinitionPrivate {
         value: revoc_key_priv,
     };
 
     trace!(
-        "create_revocation_registry <<< revoc_reg_def: {:?}, private: {:?}, revoc_reg: {:?}",
+        "create_revocation_registry <<< revoc_reg_def: {:?}, private: {:?}",
         revoc_reg_def,
         secret!(&revoc_def_priv),
-        revoc_reg
     );
 
-    Ok((revoc_reg_def, revoc_def_priv, revoc_reg, revoc_init_delta))
+    Ok((revoc_reg_def, revoc_def_priv))
 }
 
-pub fn update_revocation_registry(
+pub fn create_revocation_status_list(
+    rev_reg_def_id: impl TryInto<RevocationRegistryDefinitionId, Error = ValidationError>,
     rev_reg_def: &RevocationRegistryDefinition,
-    rev_reg: &RevocationRegistry,
-    issued: BTreeSet<u32>,
-    revoked: BTreeSet<u32>,
-    tails_reader: &TailsReader,
-) -> Result<(RevocationRegistry, RevocationRegistryDelta)> {
-    let mut rev_reg = rev_reg.value.clone();
-
+    timestamp: Option<u64>,
+    issuance_by_default: bool,
+) -> Result<RevocationStatusList> {
+    let mut rev_reg: ursa::cl::RevocationRegistry = serde_json::from_str(ACCUM_NO_ISSUED)?;
     let max_cred_num = rev_reg_def.value.max_cred_num;
-    let delta = CryptoIssuer::update_revocation_registry(
-        &mut rev_reg,
-        max_cred_num,
-        issued,
-        revoked,
-        tails_reader,
-    )?;
-    Ok((
-        RevocationRegistry { value: rev_reg },
-        RevocationRegistryDelta { value: delta },
-    ))
+    let validated_rev_reg_def_id = rev_reg_def_id.try_into()?;
+    validated_rev_reg_def_id.validate()?;
+
+    let list = if issuance_by_default {
+        let tails_reader = TailsFileReader::new_tails_reader(&rev_reg_def.value.tails_location);
+        let issued = BTreeSet::from_iter(1..=max_cred_num);
+
+        CryptoIssuer::update_revocation_registry(
+            &mut rev_reg,
+            max_cred_num,
+            issued,
+            BTreeSet::new(),
+            &tails_reader,
+        )?;
+        bitvec![0; max_cred_num as usize ]
+    } else {
+        bitvec![1; max_cred_num as usize ]
+    };
+
+    RevocationStatusList::new(
+        Some(validated_rev_reg_def_id.to_string().as_str()),
+        list,
+        Some(rev_reg),
+        timestamp,
+    )
+}
+
+/// Update Revocation Status List
+/// - if `rev_reg_def` is `Some`: updates will be made to the accumulator and status_list bitvec
+/// according to the `BTreeSet`
+/// - if `rev_reg_def` is `None`: only the timestamp will be updated
+pub fn update_revocation_status_list(
```

Comment:
if the issuer pass in `None` for `rev_reg_def` then only the `timestamp` will be updated, for timestamp created to match the ledger.

Reply:
Would it make sense, later, to extract this into two methods? If you want to update the timestamp you have to supply 3 times `None`.